### ISSUE TITLE
Download Issue

### ISSUE THREAD

**Issue by asadkhanmaharvi** [YifanDengWHU/DDIMDL#5]
I am running the **DRKG_drug_spider.py** file but always it is printing 150???

**Comment by YifanDengWHU** [comment]
Hi! I just debug the code and found that there may be something wrong with the parser of BeautifulSoup package. I changed the way from 'lxml' to 'html.parser' just now and it works. I also updated the annotation of the code, and hope it will help.
Also, the spider is not always a stable way. You can check your last issue, the worker from DrugBank said that we can parse the official data.

**Comment by asadkhanmaharvi** [comment]
Yes, I also find it that you miss Html.parser, can you tell me where from you get the smile because am familiar with such type of smile 
FC(F)(F)COC1=CC(C(=O)NCC2CCCCN2)=C(OCC(F)(F)F)C=C1 not this 9|10|14|18|19|20|178|181|283|284|285|286|299|308|332|338|339|340|341|344|345|346|347|351|352|365|366|367|380|393|405|406|528|563|566|567|571|582|592|614|615|617|637|638|639|643|661|662|663|679|680|681|682|683|689|690|691|701|703 

and one thing more I think spider code needs to refine because the target is not collecting with this code am working on it maybe I find it soon and where from I can get the event number?

**Comment by YifanDengWHU** [comment]
The smiles in my database have been transformed with RDKit.  You can search how to use RDKit to transform the SMILES into fingerprints. Here I transform SMILES into 881 dimension vectors.
For the second question, I think you are talking about pathway rather than target, right? (Because target is obtained in Line 129 of the code ) In fact, the pathway denotes the position related to the drugs' target, enzyme, carriers, and transporters.  You can see the uniprot ID of the target/enzyme in the drugbank website. We used the uniprot ID and obtain its related pathways from KEGG. In fact, the result shows that the pathway didn't help much in the experiment, so we didn't carry further research.
As for the event number, you can simply replace the drugs' names in event description with DrugA and DrugB. And then use the select count(*) from XXX(table name) group by event. So that you can tag event number for the events.

**Comment by asadkhanmaharvi** [comment]
Thanks for consideration can you guide me what is an exact prediction in it, I have print and check that [3. 2. 0. ... 2. 0. 0.] I am getting it as a prediction case or y_true. is it due to one-hot encoding?

**Comment by YifanDengWHU** [comment]
The original prediction has the shape of [batch_size, event_number], and it corresponds to the shape of y_true_one_hot. 
So I think [3. 2. 0. ... 2. 0. 0.] may be y_true or pred_type. They both have the shape of [batch_size, 1] or just [batch_size]. 
pred_type is obtained by np.argmax(prediction, axis=1).

**Comment by asadkhanmaharvi** [comment]
i want to know that what we predict as you have said that it is due to batch size it means real one is 3 if I talk about for only one so this 3 means mechanism and action?

**Comment by asadkhanmaharvi** [comment]
I just want to explore it on one drug as x_test means how it is working?

**Comment by YifanDengWHU** [comment]
Yes, after you get the prediction result 3 for a drug-drug pair. Then you have look up the mapping relationship between your labels and mechanism, action. So that you can know the potential DDI event between the pair.

**Comment by asadkhanmaharvi** [comment]
can we discuss it in depth with another platform rather than posting here?

**Comment by YifanDengWHU** [comment]
Leave your IM app(telegram, what’s app etc) and account and I’ll contact you there.

**Comment by asadkhanmaharvi** [comment]
+923004146023

**Comment by YifanDengWHU** [comment]
Hi! Which IM app do you use? I didn't find you on telegram.

**Comment by asadkhanmaharvi** [comment]
Whatsapp +923004146023

**Comment by asadkhanmaharvi** [comment]
> The smiles in my database have been transformed with RDKit. You can search how to use RDKit to transform the SMILES into fingerprints. Here I transform SMILES into 881 dimension vectors.

Are you talking about smiles to MOL ??
